src/utils.js

```diff
@@ -314,10 +314,21 @@ export function optionsList(schema) {
 
 function findSchemaDefinition($ref, definitions = {}) {
   // Extract and use the referenced definition if we have it.
-  const match = /#\/definitions\/(.*)$/.exec($ref);
-  if (match && match[1] && definitions.hasOwnProperty(match[1])) {
-    return definitions[match[1]];
+  const match = /^#\/definitions\/(.*)$/.exec($ref);
+  if (match && match[1]) {
+    const parts = match[1].split("/");
```

Comment:
@n1k0: Looking at the [JSON Pointer spec, §3. Syntax](https://tools.ietf.org/html/rfc6901#section-3), names containing `/` or `~` will be escaped as `~1` and `~0` respectively.  I will update to handle this, but I can't help wondering whether this project might instead use something like [json-schema-ref-parser](https://github.com/BigstickCarpet/json-schema-ref-parser) to handle these and other issues?